Testscripts/Windows/PERF-XDP-NTTTCP.ps1

```diff
@@ -0,0 +1,217 @@
+        $currentTestResult.TestSummary += New-ResultSummary -testResult $thrWithtXDP -metaData "With XDP Throughput $([math]::Pow(2,$itr)) connections" -checkValues "PASS,FAIL,ABORTED" -testName $currentTestData.testName
+        $thresholdLimit = [math]::Floor([double]($thrWithoutXDP * $thresholdThroughput))
+        if ($thrWithtXDP -lt $thresholdLimit) {
+            Write-LogErr "Throughput With XDP running $ThrWithXDP is less than `
+                threshold % of throughput without XDP $thrWithoutXDP"
+            $testFailedFlag = $true
+        }
+        else {
+            Write-LogInfo "Throughput with XDP running $ThrWithXDP is `
+                greater than threshold % of throughput without XDP $thrWithoutXDPs"
+        }
+
+    }
+    if ($testFailedFlag) {
+        return $false
+    }
+    else {
+        return $true
+    }
+}
+
+function Main {
+    try {
+        $noReceiver = $true
+        $noSender = $true
+        foreach ($vmData in $allVMData) {
+            if ($vmData.RoleName -imatch "receiver") {
+                $receiverVMData = $vmData
+                $noReceiver = $false
+            }
+            elseif ($vmData.RoleName -imatch "sender") {
+                $noSender = $false
+                $senderVMData = $vmData
+            }
+        }
+        if ($noReceiver) {
+            Throw "No Receiver VM defined. Aborting Test."
+        }
+        if ($noSender) {
+            Throw "No Sender VM defined. Aborting Test."
+        }
+
+        #CONFIGURE VM Details
+        Write-LogInfo "CLIENT VM details :"
+        Write-LogInfo "  RoleName : $($receiverVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($receiverVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($receiverVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($receiverVMData.InternalIP)"
+        Write-LogInfo "SERVER VM details :"
+        Write-LogInfo "  RoleName : $($senderVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($senderVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($senderVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($senderVMData.InternalIP)"
+
+        # PROVISION VMS FOR LISA WILL ENABLE ROOT USER AND WILL MAKE ENABLE PASSWORDLESS AUTHENTICATION ACROSS ALL VMS.
+        Provision-VMsForLisa -allVMData $allVMData -installPackagesOnRoleNames "none"
+
+        Write-LogInfo "Generating constants.sh ..."
+        $constantsFile = "$LogDir\constants.sh"
+        Set-Content -Value "#Generated by Azure Automation." -Path $constantsFile
+        Add-Content -Value "ip=$($receiverVMData.InternalIP)" -Path $constantsFile
+        Add-Content -Value "client=$($receiverVMData.InternalIP)" -Path $constantsFile
+        Add-Content -Value "server=$($senderVMData.InternalIP)" -Path $constantsFile
+        Add-Content -Value "testServerIP=$($senderVMData.SecondInternalIP)" -Path $constantsFile

```

Comment:
it's a little bit complicated design, especially to change the ntttcp_perf.sh, which should be a stable and common file to be used by wide test cases.   Why not enable/disable XDP from one of server/sender and then Run ntttcp again, and compare the result?

Reply:
As discussed over teams call. I have made changes.